flink-dist/pom.xml

```diff
@@ -34,7 +34,7 @@ under the License.
 	<packaging>jar</packaging>
 
 	<properties>
-		<zookeeper.optional.version>3.5.6</zookeeper.optional.version>
+		<zookeeper.optional.version>3.5.9</zookeeper.optional.version>
```

Comment:
I'm curious - why was the optional zookeeper version bumped as part of the flink-shaded bump?

Reply:
This references the version of flink-shaded-zookeeper.
We have 2 artifacts of that; one for ZK 3.4, one for 3.5 . 3.4 is added to the lib/ as the default, 3.5 to opt/.

The versions for all flink-shaded artifacts contain both the flink-shaded version (14.0) and the version of the bundled dependency (3.5.9). Since flink-shaded 1.14 contains an upgrade version of ZK 3.5, we now also have to update this property.